lib/rubocop/cop/style/empty_case_condition.rb

```diff
@@ -70,21 +70,24 @@ def correct_case_whens(corrector, case_node, when_nodes)
         end
 
         # Since an if condition containing commas is not syntactically valid, we
-        # correct `when x,y` to `if [x,y].any?`.
+        # correct `when x, y` to `if x || y`.
         def correct_multiple_alternative_whens(corrector, when_nodes)
           when_nodes.each do |when_node|
-            children = when_node.children
+            *children, _ = when_node.children
 
             # In `when a; r` we have two children: [a, r].
             # In `when a, b, c; r` we have 4.
-            next unless children.size > 2
+            # In `when a, b then r` we have 3.
+            next unless children.size > 1
 
+            first = children.first
+            last = children.last
             range =
               Parser::Source::Range.new(when_node.loc.expression.source_buffer,
-                                        children[0].loc.expression.begin_pos,
-                                        children[-2].loc.expression.end_pos)
+                                        first.loc.expression.begin_pos,
+                                        last.loc.expression.end_pos)
 
-            corrector.replace(range, children[0..-2].map(&:source).join(' || '))
+            corrector.replace(range, children[0..-1].map(&:source).join(' || '))
```

Comment:
Yes, it's definitely not a real world example :-) but the `when`s *can* contain literals, even if it doesn't make sense for them to do so. 

Since it's possible to write code that is accepted by MRI, that *will* be broken by this cop's autocorrection I wonder if we should try and handle this case (by wrapping the components with parens) - it would be bad to break some deeply-nested, rarely-used code that wouldn't be discovered for a long time! Perhaps we could conditionally add the parens, since it would be ugly to add them in the general case.

I'm not sure what the general approach of Rubocop is for such unlikely-but-possible cases? I'm fine with leaving this as-is, if the consensus is that we're ok with rare cases being broken.